core/services/job/models.go

```diff
@@ -39,20 +39,33 @@ func (t Type) String() string {
 	return string(t)
 }
 
-func (t Type) HasPipelineSpec() bool {
-	return hasPipelineSpec[t]
+func (t Type) RequiresPipelineSpec() bool {
+	return requiresPipelineSpec[t]
+}
+
+func (t Type) SupportsAsync() bool {
+	return supportsAsync[t]
 }
 
 var (
-	hasPipelineSpec = map[Type]bool{
+	requiresPipelineSpec = map[Type]bool{
 		Cron:              true,
 		DirectRequest:     true,
 		FluxMonitor:       true,
-		OffchainReporting: true,
+		OffchainReporting: false, // bootstrap jobs do not require it
 		Keeper:            false,
 		VRF:               true,
 		Webhook:           true,
 	}
+	supportsAsync = map[Type]bool{
+		Cron:              false,
+		DirectRequest:     false,
```

Comment:
Pretty sure DirectRequest also supports async, no?

Reply:
It doesn't yet, I'm working on that right now.